test/index.test.js

```
@@ -110,6 +110,25 @@ describe('FullStory', function() {
         analytics.identify('id3', { name: 'Steven', registered: true });
         analytics.called(window.FS.identify, 'id3', { displayName: 'Steven', registered_bool: true });
       });
+
+      it('should map arrays properly', function() {
+        analytics.identify('id3', { teams: ['eng', 'redsox'] });
+        analytics.called(window.FS.identify, 'id3', { teams: [ 'eng', 'redsox'] });
+      });
+
+      it('should map user objects properly', function() {
+        analytics.identify('id3', { account: { level: 'premier', avg_annual: 30000 } });
+        analytics.called(window.FS.identify, 'id3', { account: { level: 'premier', avg_annual: 30000 } });
+      });
+
+      it('should respect existing type tags', function() {
+        // All of these would break FS if passed; they prove that if the user has a
+        // variable of type X, but pass a type Y, we use their name for the variable.
+        analytics.identify('id3', { my_int_str: 17, my_str_int: 'foo', my_int_date: 3,
+               my_int_bool: 4, mystr_real: 'plugh' });
+        analytics.called(window.FS.identify, 'id3', { myInt_str: 17, myStr_int: 'foo',
```

This is the objection you recanted, right?

I'm trying to catch a couple cases here.  The first, which I think you now grok, is that if they use e.g. varname_int but pass a string, Segment shouldn't blindly create varnameInt_str because of what the value was; varname_int should pass through, with the string, and we'll whine about that mismatch when we get it.  Segment shouldn't mask the error.

The other cute case is if the user passes varname_int and corectly pass an _int_, we don't want Segment creating varnameInt_int.  We want things that smell like they have type qualifiers not to be messed with.  Probably at all, but certainly not much.  (I did keep Segment's camel-case function for the varname body; maybe I shouldn't have.)